mne/viz/utils.py

```diff
@@ -60,20 +60,22 @@ def _setup_vmin_vmax(data, vmin, vmax, norm=False):
     return vmin, vmax
 
 
-def plt_show(show=True, **kwargs):
+def plt_show(show=True, fig=None, **kwargs):
     """Show a figure while suppressing warnings.
 
     Parameters
     ----------
     show : bool
         Show the figure.
+    fig : instance of Figure | None
+        If non-None, use fig.show().
     **kwargs : dict
         Extra arguments for :func:`matplotlib.pyplot.show`.
     """
     import matplotlib
     import matplotlib.pyplot as plt
     if show and matplotlib.get_backend() != 'agg':
-        plt.show(**kwargs)
+        (fig or plt).show(**kwargs)
```

Comment:
wtf is this python voodoo...

Reply:
Python has a short-circuit functionality. So if `fig` evaluates to True (in this case if it's a `Figure` instead of the default `None`) it will use it, otherwise it will use `plt`

Reply:
that is magical